hibernate-core/src/main/java/org/hibernate/loader/entity/UniqueEntityLoader.java

```diff
@@ -60,4 +69,11 @@ default Object load(
 			Boolean readOnly) {
 		return load( id, optionalObject, session, lockOptions );
 	}
+
+	default Object load(
+			Object id,
+			SharedSessionContractImplementor session,
+			LockOptions lockOptions) {
+		throw new UnsupportedOperationException();
+	}
```

Comment:
This is a replacement for `loadByUniqueKey()`. Unimplemented by default. But in future we could extend unique key loading to other `UniqueEntityLoader`s.